### PR TITLE
Use test-beat role for all Beats

### PR DESCRIPTION
**Pull request by andrewkroh**
Update Metricbeat, Heartbeat, Filebeat, and Winlogbeat to use the test-beat roles.

Update run-settings for staging to 6.2.0 BC3.

Copy logs/config/output to host even on error. This uses a task block with an "always" condition (like a try/finally) in order to ensure that log files, output files, and config are copied to the localhost.

Add workaround for https://github.com/elastic/beats/issues/6214 on Darwin.

### CODE REVIEW

**Comment by tsg** [comment]
I was trying it locally and got this error:

```
TASK [test-beat : Create log file] *********************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AnsibleFileNotFound: Could not find or access '/Users/tsg/src/github.com/elastic/beats-tester/roles/test-beat/files/filebeat-test.log'
```

The file seems to be missing, or at least it doesn't seem to be in another path according to my greps.

**Comment by andrewkroh** [comment]
@tsg I forgot to git add the file. It's pushed now.

**Comment by ruflin** [comment]
Nice change. Removes quite a bit of duplicated code and makes things simpler 🎉 

Will not merge it as I can't test this locally currently. I'll leave it to @tsg ;-)

File: roles/test-beat/tasks/darwin/main.yml [hunk 1]
```diff
@@ -25,14 +25,20 @@
     dest: '{{ beat_cfg }}'
     mode: 0600
 
+# TODO (andrewkroh on 2018-01-29): Remove -path.logs after resolving
+# https://github.com/elastic/beats/issues/6214.
 - name: 'Start {{ beat_service_name }} service'
-  command: './{{ beat_name }}'
+  command: './{{ beat_name }} -path.logs={{ beat_logs_path }}'
```
Comment (ph) [inline roles/test-beat/tasks/darwin/main.yml]: @andrewkroh You can remove the workaround for elastic/beats#6214

Comment (andrewkroh) [reply roles/test-beat/tasks/darwin/main.yml]: Thanks for the quick fix. The work-around has been removed.

Comment (ph) [reply roles/test-beat/tasks/darwin/main.yml]: @andrewkroh FYI, it will work with BC5, when it happens.

**Comment by andrewkroh** [comment]
If it doesn't hurt anything then I would put it into the Makefile. Otherwise (or if we aren't sure) the readme.

**Pull request merged**